DscResources/MSFT_xMySqlServer/MSFT_xMySqlServer.psm1

```diff
@@ -1,303 +1,112 @@
+        $ensureResult = "Absent"
+        $portResult = $null
     }
-}
-
-################################################################################################################################
-# Get-MySqlProductName (helper to get product name of MySQL)
-################################################################################################################################
 
-function Get-MySqlProductName
-{
-    return "mysql-server-$mySqlVersion-$(Get-MySqlArchitectureName)"
-}
-
-################################################################################################################################
-# Get-MySqlCatalogName (helper to get the MySQL catalog name)
-################################################################################################################################
-
-function Get-MySqlCatalogName
-{
-    return "mysql-$mySqlVersion-$(Get-MySqlArchitectureName)"
+    return @{
+        Ensure       = $ensureResult
+        MySqlVersion = $MySqlVersion
+        Port         = $portResult
+    }
 }
 
-#endregion
-
-#region *-TargetResource implementation
-
-################################################################################################################################
-# Get-TargetResource (params: ServiceName, Ensure, RootPassword)
-################################################################################################################################
-
-function Get-TargetResource
+function Set-TargetResource 
 {
-    [OutputType([Hashtable])]
     param
     (
         [parameter(Mandatory = $true)]
-        [string] $ServiceName,
-        
+        [ValidateNotNullOrEmpty()]
+        [string] $MySqlVersion,
+
         [ValidateSet("Present", "Absent")]
         [string] $Ensure = "Present",
 
         [parameter(Mandatory = $true)]
-        [pscredential] $RootPassword
+        [pscredential] $RootPassword,
+
+        [string] $Port = "3306"
+        
     )
     
-    $Ensure = "Absent"
-    $MySqlInstalled = $false 
-
-    # get installed products
-    $products = @(Get-MySqlInstalledComponents)
-
-    if ($products -ne $null)
+    if ($Ensure -eq "Present")
     {
-        foreach($product in $products)
-        {
-            Trace-Message  "'$product' is currently installed"
-        }
-    
-        Trace-Message "Checking to see if installed products contains '$((Get-MySqlProductName))'"
+        $versionsInstalled = Get-MySqlAllInstalled
 
-        if($products -contains (Get-MySqlProductName))
+        if ($versionsInstalled -eq $null)
         {
-            $MySqlInstalled  = $true 
+            Write-Verbose "Installing MySQL..."
+            $arguments = "community", "install", `
+                "server;$MySqlVersion;$(Get-ArchitectureName):*:servertype=Server;port=$Port;passwd=$($RootPassword.GetNetworkCredential().Password)", "-silent"
+            $null = Invoke-MySqlCommand -CommandPath $(Get-MySqlInstallerConsole) -Arguments $arguments
```

Comment:
What I meant was, it would only install one service using the installer's default service name for a given version. With this change you would not be able to install multiple instances of the same version and run them as different services at the same time as it used to do before.